src/plugins/recursion/README.md

```diff
@@ -0,0 +1,64 @@
+
+
+## Introduction
+
+I took [Relax NG](http://relaxng.org/spec-20011203.html#define-ref)'s `ref` and `define` part and tried to "copy" 
+their approach for elektra's specification.
+Until now I haven't really gotten a sensible idea for that though. Maybe we could talk on the next meeting about this.
+
+Also I have no idea what your `struct` plugin actually does. There is no example, just some code snippet from 
+the contract which doesn't help understanding the plugin.
+
+## Problems
+1) We want to check if all recursions are non-cyclical
+2) We want to check if a recursive config actually leads to another correct config and not into NULL for example
+3) Forcing users to define non-recursive settings first on which they can build up 
+recursive settings is inconvenient and could lead to orphan settings.
+4) We can only check right after calling `kdb set` and not a whole config. 
+This means that user can have "unfinished" configurations while setting new values.
```

Comment:
I do not understand this limitation. Within "kdb set" you can check the whole configuration and reject everything that is unfinished.

Reply:
That's correct, but this would force the user to have a specified ordering when creating the configuration. Either "bottom-up" (you have to create leaf settings before recursive ones) or "top-down" (you have to create top level settings first before adding lower level ones).

Klemens said that some plugins which take a config do not have a specific ordering once you load them.

Reply:
Some storage plugins do not remember the order of entries *within the configuration files*. But the order of the configuration structure should have nothing to do with that.